tests/integration/web/maintenance/utils_test.py

```diff
@@ -0,0 +1,29 @@
+from nav.models.manage import Netbox, Room
+from nav.web.maintenance.utils import prefetch_and_group_components
+
+
+class TestPrefetchAndGroupComponents:
+    def test_groups_netboxes_by_room(self, db, client):
+        netbox = Netbox.objects.first()
+        netbox_query = Netbox.objects.filter(id=netbox.id)
+        results = prefetch_and_group_components(Netbox, netbox_query, Room)
+        assert results, "No results returned from prefetch_and_group_components."
+        assert all(isinstance(item, tuple) for item in results), (
+            "Results should be tuples."
+        )
+        room_name, netbox_list = results[0]
+        assert room_name == str(netbox.room), "Room name does not match Netbox's room."
+        assert netbox.id == netbox_list[0].id, (
+            "Netbox ID does not match expected value."
+        )
```

Comment:
Sometimes, multiple assertions in a single test indicate that you might actually want multiple tests (the ideal number of assertions in a single test is 1, though that isn't always achievable, of course).

When I see a report where test named `TestPrefetchAndGroupComponents.test_should_group_netboxes_by_room` failed, I would ideally like to be able to grok from its name what kind of expectation failed, yet this test seems to test about four different sub-aspects of the function's return value.